game-app/game-core/src/main/java/games/strategy/triplea/odds/calculator/AggregateResults.java

```diff
@@ -170,20 +284,66 @@ public double getAverageTuvSwing(
     return count / total;
   }
 
+  /** Returns the standard deviation of the defending units left if the defender won. */
+  double getAverageDefendingUnitsLeftWhenDefenderWonStandardDeviation() {
+    if (results.isEmpty()) {
+      return 0.0;
+    }
+    final double mu = getAverageDefendingUnitsLeftWhenDefenderWon();
+    double count = 0;
+    double total = 0;
+    for (final BattleResults result : results) {
+      if (result.defenderWon()) {
+        count += Math.pow(result.getRemainingDefendingUnits().size() - mu, 2);
+        total += 1;
+      }
+    }
+    if (total <= 0) {
+      return 0;
+    }
+    return Math.sqrt(count / total);
+  }
+
   public double getAttackerWinPercent() {
     if (results.isEmpty()) {
       return 0.0;
     }
     return results.stream().filter(BattleResults::attackerWon).count() / (double) results.size();
   }
 
+  /** Returns the standard deviation of the attacker wins percentage. */
```

Comment:
The std deviation for number of times the battle is won is seemingly very helpful. This is a good addition :+1:

Reply:
Funny, I think that this std dev is the least useful of all the added ones (I even considered not adding it) because it only depends on the average (=win percentage), thus does not add any information.  For TUV swing etc the std dev really adds something not already covered by the expected value/average.